src/arch/riscv/irqc/aia/inc/irqc.h

```diff
@@ -0,0 +1,53 @@
+#ifndef _IRQC_H_
+#define _IRQC_H_
+
+#include <aplic.h>
+#include <cpu.h>
+#include <vaplic.h>
+
+#define XPLIC_MAX_INTERRUPTS (1024)
+
+#define HART_REG_OFF APLIC_IDC_OFF
+#define IRQC_HART_INST APLIC_PLAT_IDC_NUM
+
+static inline void irqc_init()
+{
+    aplic_init();
+}
+
+static inline void irqc_cpu_init()
+{
+    aplic_idc_init();
+}
+
+static inline void irqc_set_enbl(irqid_t int_id, bool en)
+{
+    aplic_set_ienum(int_id);
+}
+
+static inline void irqc_set_prio(irqid_t int_id)
+{
+    aplic_set_target(int_id, (cpu()->id << 18) | 0x01);
+}
+
+static inline void irqc_handle()
+{
+    aplic_handle();
+}
+
+static inline bool irqc_get_pend(irqid_t int_id)
+{
+    return aplic_get_pend(int_id);
```

Comment:
I'm not sure if I understand. The call of this function should return whether a particular interrupt is pending in this domain. Whether it is pending will depend on factors like whether it is active or not, and the configuration mode in sourcecfg.SM. So, an interrupt may be at 1 in the rectification register, but it may not mean it is pending.